src/model/Service.php

```diff
@@ -1,37 +1,47 @@
 <?php
 namespace base\model;
 
+use base\model\entity\EntityFactory;
+use base\model\dataMapper\DataMapperFactory;
+
 class Service
 {	
 	protected $dataMapper;
 	protected $entityFactory;
-	protected $entityType;
 	
-	public function __construct($dataMapper,$entityType)
+	public function __construct($modelConfig)
     {
-		$this->dataMapper = $dataMapper;
-		$this->entityType = $entityType;
+		$dataMapperFactory = new DataMapperFactory($modelConfig["dataConnection"]);
+		$this->dataMapper = $dataMapperFactory->getDataMapper();
 	}
 	
-	public function buildEntity( $name ){
-		switch($this->entityType){
-			case "autoGen":
-				$this->entityFactory = new AutoGenEntityFactory();
-				$attributes = "";
-				foreach($this->dataMapper->fetchColumns($name) as $id => $name)$attributes[$name] = "";
-				return $this->entityFactory->getEntity($name,$attributes);
-				break;
-			case "normal":
-				$this->entityFactory = new EntityFactory();
-				return $this->entityFactory->getEntity($name);
-				break;
-		}
+	public function buildEntity( $name )
+	{
+		$this->entityFactory = new EntityFactory();
+		$entity = $this->entityFactory->getEntity($name);
+		
+		//si l'entity est "vide"
+		if(is_null($entity->getRepositoryName())){
```

Comment:
If I want to load an entity schema from XML or config, I have to add postProcess here? Too heavy and not maintainable. Because, if I want to remove autoGenEntity, I will have to find all code related to this specific case. I don't want to play at 'where is Charly?'